modules/manually-restoring-cluster-etcd-backup.adoc

```diff
@@ -0,0 +1,312 @@
++
+[NOTE]
+====
+It is not required to manually stop the pods on the recovery host. The recovery script will stop the pods on the recovery host.
+====
+
+.. Access a control plane host that is not the recovery host.
+
+.. Move the existing Kubernetes API Server static pod manifest out of the kubelet manifest directory
++
+[source,terminal]
+----
+$ sudo mv /etc/kubernetes/manifests/kube-apiserver-pod.yaml /tmp
+----
+
+.. Verify that the Kubernetes API Server containers have stopped:
++
+[source,terminal]
+----
+$ sudo crictl ps | grep kube-apiserver | grep -v "operator|gaurd"
+----
++
+The output of this command should be empty. If it is not empty, wait a few minutes and check again.
+
+.. If Kubernetes API Server containers haven't stopped yet, force their termination with:
++
+[source,terminal]
+----
+$ sudo crictl stop <container_id>
+----
+
+.. Repeat the same steps for `kube-controller-manager-pod.yaml`, `kube-scheduler-pod.yaml` and eventually `etcd-pod.yaml`.
+
+.. On each control plane host, move the current `etcd` data into the `backup` folder:
++
+[source,terminal]
+----
+$ sudo mv /var/lib/etcd/member /home/core/assets/backup/
+----
++
+This data will be useful in case the `etcd` backup restore doesn't work and the `etcd` cluster must be restored to the current state.
+
+. Find the correct etcd parameters for each control plane host
+
+.. The value for `<ETCD_NAME>` is related to the current host and it is equal to the value of the `ETCD_NAME` variable in the manifest `/etc/kubernetes/static-pod-resources/etcd-certs/configmaps/restore-etcd-pod/pod.yaml` in the current host. It can be found with:
++
+[source,terminal]
+----
+$ grep -A 1 \
+  $(grep 'export ETCD_NAME' /etc/kubernetes/static-pod-resources/etcd-certs/configmaps/restore-etcd-pod/pod.yaml | sed -E 's/.*\{(.*)\}/\1/g') \
+  /etc/kubernetes/static-pod-resources/etcd-certs/configmaps/restore-etcd-pod/pod.yaml | \
+  grep "value" | \
+  sed -E 's/.*\"(.*)"/\1/g'
+----
+
+.. The value for `<UUID>` can be generated with the command:
++
+[source,terminal]
+----
+$ uuidgen
+----
++
+[NOTE]
+====
+Note that `<UUID>` must be generated only once and the same value must used on all the control plane hosts.
+====
+
+.. The value for `ETCD_NODE_PEER_URL` should be like the following:
++
+[source,yaml]
+----
+https://<IP_CURRENT_HOST>:2380
+----
++
+The correct IP can be found with the following command, using the `<ETCD_NAME>` of the current host:
++
+.Example output
+[source,terminal]
+----
+ETCD_NAME=<ETCD_NAME> && grep -A 1 "export NODE"  /etc/kubernetes/static-pod-resources/etcd-certs/configmaps/etcd-scripts/etcd.env | \
+  grep -A 1 "$ETCD_NAME" | \
+  grep "ETCD_URL_HOST" |  \
+  sed -E 's/.*\"(.*)\"/\1/g'
+----
+
+.. The value for `<ETCD_INITIAL_CLUSTER>` should be like the following, where `<ETCD_NAME_n>` is the `<ETCD_NAME>` of each control plane host. 
++
+[NOTE]
+====
+Note that the port used here is 2380 and not 2379.
```

Comment:
@vwalek ^

Reply:
Any update from @vwalek?

Reply:
>because the default port is 2379?
Actually, the etcd sets different ports than 2379/2380 as default. I think default is 4001. Nevertheless, it would be good to say that Kubernetes etcd instance configures ports 2380 for data exchange, 2379 for management. The port 2379 is added to etcd when it starts. 
Would be good to rephrase it as.

The port used in recovery must be 2380. (The port 2379 is used for `etcd` database management and is configured directly in `etcd` start command in container)

Reply:
@acancell-redhat as mentioned above, I will rephrase the sentence.